index.d.ts

```diff
@@ -1,294 +1,276 @@
-        BeforeCreateScheduleEventHandlerFunc |
-        BeforeDeleteScheduleEventHandlerFunc |
-        BeforeUpdateScheduleEventHandlerFunc |
-        ClickDayNameEventHandlerFunc |
-        ClickMoreEventHandlerFunc |
-        ClickScheduleEventHandlerFunc |
-        TimezonesCollapseEventFunc;
-    type CustomEventType = 'afterRenderSchedule' | 'beforeCreateSchedule' |
-        'beforeDeleteSchedule' | 'beforeUpdateSchedule' | 'clickDayname' |
-        'clickMore' | 'clickSchedule' | 'clickTimezonesCollapseBtn';
+export type DateType = string | Date | TZDate;
+export type EventHandlerType = IEvents[keyof IEvents];
+export type CustomEventType = keyof IEvents;
 
-    interface IEventObject {
-        schedule: ISchedule;
-        end: TZDate;
-        start: TZDate;
-        calendar?: ICalendarInfo;
-        triggerEventName?: string;
-    }
+export interface IEventObject {
+    schedule: ISchedule;
+    end: TZDate;
+    start: TZDate;
+    calendar?: ICalendarInfo;
+    triggerEventName?: string;
+}
 
-    interface IEventDateObject {
-        date: string;
-    }
+export interface IEventDateObject {
+    date: string;
+}
 
-    interface IEventMoreObject {
-        date: TZDate;
-        target: Element;
-    }
+export interface IEventMoreObject {
+    date: TZDate;
+    target: Element;
+}
 
-    interface IEventScheduleObject {
-        calendar: ICalendarInfo;
-        event: MouseEvent;
-        schedule: ISchedule;
+export interface IEventScheduleObject {
+    calendar: ICalendarInfo;
+    event: MouseEvent;
+    schedule: ISchedule;
+}
 
-    }
-    interface IEvents {
-        'afterRenderSchedule'?: AfterRenderScheduleEventHandlerFunc;
-        'beforeCreateSchedule'?: BeforeCreateScheduleEventHandlerFunc;
-        'beforeDeleteSchedule'?: BeforeDeleteScheduleEventHandlerFunc;
-        'beforeUpdateSchedule'?: BeforeUpdateScheduleEventHandlerFunc;
-        'clickDayname'?: ClickDayNameEventHandlerFunc;
-        'clickMore'?: ClickMoreEventHandlerFunc;
-        'clickSchedule'?: ClickScheduleEventHandlerFunc;
-        'clickTimezonesCollapseBtn'?: TimezonesCollapseEventFunc;
-    }
+export interface IEvents {
+    'afterRenderSchedule'?: (eventObj: {schedule: ISchedule}) => void;
+    'beforeCreateSchedule'?: (schedule: ISchedule) => void;
+    'beforeDeleteSchedule'?: (eventObj: IEventScheduleObject) => void;
+    'beforeUpdateSchedule'?: (eventObj: IEventObject) => void;
+    'clickDayname'?: (eventObj: IEventDateObject) => void;
+    'clickMore'?: (eventObj: IEventMoreObject) => void;
+    'clickSchedule'?: (eventObj: IEventScheduleObject) => void;
+    'clickTimezonesCollapseBtn'?: (timezonesCollapsed: boolean) => void;
+}
 
-    class TZDate {
-        public getTime(): number;
-        public toDate(): Date;
-        public toUTCString(): Date;
-    }
+export class TZDate {
+    public getTime(): number;
+    public toDate(): Date;
+    public toUTCString(): Date;
```

Comment:
This return type must be `string`. And check and fix the JSDoc at timezone.js please.

Reply:
right. The return type of toUTCString is `string`.
doc fixed. too